iothub/device/src/Convention/CommandRequest.cs

```diff
@@ -42,17 +43,32 @@ internal CommandRequest(PayloadConvention payloadConvention, string commandName,
         public string CommandName { get; }
 
         /// <summary>
-        /// The command request payload.
+        /// The command request payload, deserialized to the specified type.
         /// </summary>
         /// <typeparam name="T">The type to deserialize the command request payload to.</typeparam>
-        /// <returns>The command request payload.</returns>
-        public T GetPayload<T>()
+        /// <param name="payload">When this method returns true, this contains the value of the command request payload.
+        /// When this method returns false, this contains the default value of the type <c>T</c> passed in.</param>
+        /// <returns><c>true</c> if the command request payload can be deserialized to type <c>T</c>; otherwise, <c>false</c>.</returns>
+        public bool TryGetPayload<T>(out T payload)
```

Comment:
Updated this to a "TryGet" method, for consistency with the "TryGet" methods in properties classes.